extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -721,13 +730,55 @@ export class AmpAnalytics extends AMP.BaseElement {
   expandAndSendRequest_(request, trigger, event) {
     this.config_['vars']['requestCount']++;
     const expansionOptions = this.expansionOptions_(event, trigger);
+    /** @type {!Object<string, (DynamicVariableBindingDef|HtmlAttrBindingDef)>} */
     const dynamicBindings =
         this.getDynamicVariableBindings_(trigger, expansionOptions);
+    dynamicBindings['HTML_ATTR'] = this.htmlAttrBinding_.bind(this);
     request.send(
         this.config_['extraUrlParams'], trigger, expansionOptions,
         dynamicBindings);
   }
 
+  /**
+   * Provides a binding for getting attributes from the DOM.
+   * Most such bindings are provided in src/service/url-replacements-impl, but
+   * this one needs access to this.win.document, which if the amp-analytics
+   * tag is contained within an amp-ad tag will NOT be the parent/publisher
+   * page. Hence the need to put it here.
+   * @param {string} cssSelector Elements matching this selector will be
+   *     included, provided they have at least one of the attributeNames
+   *     set, up to a max of 10.
+   * @param attributeNames The attributes whose values will be returned.
+   * @returns {string}
+   */
+  htmlAttrBinding_(cssSelector, ...attributeNames) {
```

Comment:
Rest operator is unsupported due to polyfill requirements. /cc @erwinmombay @rsimha, this should have been caught.

Reply:
@jridgewell The only lint rule we have is the one that detects array destructuring: https://github.com/ampproject/amphtml/blob/master/build-system/eslint-rules/no-array-destructuring.js

Should it have caught this? If not, does it have to be rewritten?

Reply:
@jridgewell I think this needs a rule that guards against the use of `RestElement`, similar to the one we have in place for `SpreadElement`. See https://github.com/eslint/eslint/blob/master/lib/rules/indent.js#L64